src/scorepy/events.cpp

```diff
@@ -23,73 +23,117 @@ struct region_handle
 
-void region_begin(const std::string& region_name, std::string module, std::string file_name,
-                  std::uint64_t line_number)
+/// Region names that are known to have no region enter event and should not report an error
+/// on region exit
+static const std::array<std::string, 2> EXIT_REGION_WHITELIST = {
+#if PY_MAJOR_VERSION >= 3
+    "threading:_bootstrap_inner", "threading:_bootstrap"
+#else
+    "threading:__bootstrap_inner", "threading:__bootstrap"
+#endif
+};
+
+// Userd for regions, that have an idetifier, aka a code object id
+void region_begin(const std::string& function_name, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number,
+                  const std::uintptr_t& identifier)
 {
-    auto& region_handle = regions[region_name];
+    auto& region_handle = regions[identifier];
 
     if (region_handle == uninitialised_region_handle)
     {
-        SCOREP_User_RegionInit(&region_handle.value, NULL, &SCOREP_User_LastFileHandle,
-                               region_name.c_str(), SCOREP_USER_REGION_TYPE_FUNCTION,
-                               file_name.c_str(), line_number);
+        auto& region_name = make_region_name(module, function_name);
+        SCOREP_User_RegionInit(&region_handle.value, NULL, NULL, region_name.c_str(),
+                               SCOREP_USER_REGION_TYPE_FUNCTION, file_name.c_str(), line_number);
+
         SCOREP_User_RegionSetGroup(region_handle.value,
                                    std::string(module, 0, module.find('.')).c_str());
     }
     SCOREP_User_RegionEnter(region_handle.value);
 }
 
-/// Region names that are known to have no region enter event and should not report an error
-/// on region exit
-static const std::array<std::string, 2> EXIT_REGION_WHITELIST = {
-#if PY_MAJOR_VERSION >= 3
-    "threading:_bootstrap_inner", "threading:_bootstrap"
-#else
-    "threading:__bootstrap_inner", "threading:__bootstrap"
-#endif
-};
+// Userd for regions, that only have a function name, a module,  a file and a line number
+void region_begin(const std::string& function_name, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number)
+{
+    const auto& region_name = make_region_name(module, function_name);
+    auto& region_handle = user_regions[region_name];
+
+    if (region_handle == uninitialised_region_handle)
+    {
+        SCOREP_User_RegionInit(&region_handle.value, NULL, NULL, region_name.c_str(),
+                               SCOREP_USER_REGION_TYPE_FUNCTION, file_name.c_str(), line_number);
 
-void region_end(const std::string& region_name)
+        SCOREP_User_RegionSetGroup(region_handle.value,
+                                   std::string(module, 0, module.find('.')).c_str());
+    }
+    SCOREP_User_RegionEnter(region_handle.value);
+}
+
+// Userd for regions, that have an idetifier, aka a code object id
```

Comment:
```suggestion
// Used for regions, that have an identifier, aka a code object id (instrumenter regions)
```

Or maybe only `// instrumenter regions` as the "that have an identifier" is implied by the signature already. No need to repeat that in text as it adds no new information (IMO)